subprojects/core/src/main/java/org/gradle/api/tasks/SourceTask.java

```diff
@@ -194,4 +203,20 @@ public SourceTask setExcludes(Iterable<String> excludes) {
         patternSet.setExcludes(excludes);
         return this;
     }
+
+    /**
+     * A stable property for {@link #getSource()}.
+     *
+     * <p>
+     *     Should be used in combination with {@link org.gradle.work.InputChanges#getFileChanges(Object)} to query file changes.
+     *     For this to work, the subclass needs to override this property and add an input file annotation to it, and override {@link #getSource()} and add {@link Internal} to it.
+     * </p>
+     *
+     * @since 5.4
+     */
+    @Incubating
+    @Internal
```

Comment:
Should we mark this as `@Incremental` and `@InputFiles` and `@PathSensitive`, and mark `getSource()` as `@Internal`?

Reply:
That would possibly break the tasks which extends `SourceTask` and overwrite the path sensitivity of `getSource` with something more lenient than `@PathSensitive(RELATIVE)`.

I fail to see what the advantage of doing so would be.